nsenter/nsexec.c

```diff
@@ -103,46 +95,37 @@ void nsexec()
 		}
 	}
 
-	/* Check that the specified process exists */
-	snprintf(buf, PATH_MAX - 1, "/proc/%d/ns", pid);
-	tfd = open(buf, O_DIRECTORY | O_RDONLY);
-	if (tfd == -1) {
-		pr_perror("Failed to open \"%s\"", buf);
-		exit(1);
-	}
-
-	for (i = 0; i < num; i++) {
-		struct stat st;
-		int fd;
-
-		/* Symlinks on all namespaces exist for dead processes, but they can't be opened */
-		if (fstatat(tfd, namespaces[i], &st, AT_SYMLINK_NOFOLLOW) == -1) {
-			// Ignore nonexistent namespaces.
-			if (errno == ENOENT)
-				continue;
-		}
-
-		fd = openat(tfd, namespaces[i], O_RDONLY);
+	char *ns, *saveptr;
+	while ((ns = strtok_r(nspaths, ",", &saveptr))) {
+		fd = open(ns, O_RDONLY);
```

Comment:
If you enter into a mount namespace, you can't be sure that you will able to enter in other namespaces, because nspaths may be inaccessible from this mount namespace.